### PR TITLE
Implement sorting and filtering in "What students say" block

### LINKED ISSUE #2285: (SP: 2) Implement sorting and filtering in "What students say" block
Add the ability to sort and filter students' reviews in "What students say" block.

Blocked by #2284

### PR DESCRIPTION
**Pull request by BohdanMylyi**
Now the functionality works

Result:

https://github.com/user-attachments/assets/bdb99010-da6b-4578-a20e-e93040b9f254

### CODE REVIEW

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx [hunk 1]
```diff
@@ -55,6 +56,28 @@ const CommentsWithRatingBlock = ({
...
   ))
 
+  const filteredItems = items.filter((item) => {
+    if (filter !== null) {
+      return item.rating === filter
+    }
+    return true
+  })
+
+  const sortedItems = filteredItems.sort((a, b) => {
```
Comment (abalanovsky) [inline src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx:66]: Change to ``toSorted`` to make a copy or use filter.sort above

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx [hunk 2]
```diff
@@ -55,6 +56,28 @@ const CommentsWithRatingBlock = ({
...
+    switch (sortBy) {
+      case SortByEnum.Newest:
+        return new Date().getTime() - new Date().getTime()
```
Comment (abalanovsky) [inline src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx:69]: You are comparing two Dates that have no relations to user reviews

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx [hunk 3]
```diff
@@ -75,17 +98,37 @@ const CommentsWithRatingBlock = ({
           <Box sx={styles.container}>
             <Box sx={styles.innerBox}>
               <Typography>{t('common.labels.sortBy')}</Typography>
-              <Select defaultValue={SortByEnum.Newest}>{sortMenuItems}</Select>
+              <Select
+                defaultValue={SortByEnum.Newest}
+                onChange={(e) => {
```
Comment (abalanovsky) [inline src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx:103]: Move it to the separate function

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx [hunk 4]
```diff
@@ -75,17 +98,37 @@ const CommentsWithRatingBlock = ({
...
+                {sortMenuItems}
+              </Select>
             </Box>
 
             <Box sx={styles.innerBox}>
               <Typography>{t('common.labels.filterBy')}</Typography>
-              <Select defaultValue={5}>{ratingMenuItems}</Select>
+              <Select
+                defaultValue={5}
+                onChange={(e) =>
```
Comment (abalanovsky) [inline src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx:118]: Also move it to the separate function

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx [hunk 5]
```diff
@@ -55,6 +56,28 @@ const CommentsWithRatingBlock = ({
     </MenuItem>
   ))
 
+  const filteredItems = items.filter((item) => {
+    if (filter !== null) {
+      return item.rating === filter
+    }
+    return true
+  })
+
```
Comment (abalanovsky) [inline src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx:65]: Increase coverage

**Review by abalanovsky — CHANGES_REQUESTED** [review]

**Review by dudchakk — APPROVED** [review]

**Review by ipasic-softserve — APPROVED** [review]

**Review by docia — APPROVED** [review]

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx (hunk 1 above)
Comment (BohdanMylyi) [reply src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx]: done✅

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx (hunk 3 above)
Comment (BohdanMylyi) [reply src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx]: done✅

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx (hunk 2 above)
Comment (BohdanMylyi) [reply src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx]: already fixed

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx (hunk 4 above)
Comment (BohdanMylyi) [reply src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx]: done✅

File: src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx (hunk 5 above)
Comment (BohdanMylyi) [reply src/containers/user-profile/comments-with-rating-block/CommentsWithRatingBlock.tsx:65]: ✅

**Review by abalanovsky — APPROVED** [review]

**Pull request merged**